CONTRIBUTING.rst

```diff
@@ -38,6 +38,66 @@ General Guidelines
   from the cmd line to run the test suite).
 - Add yourself to doc/contributors.rst :)
 
+Authoring a Pull Request
+------------------------
+
+**Our Pull Request Policy is based on this** `Code Review Developer Guide <https://google.github.io/eng-practices/review>`_
+
+The expectation for any code author is to provide all the context needed in the space of a
+pull request for any engineer to feel equipped to review the code. Depending on the type of
+change, do your best to highlight important new functions or objects you’ve introduced in the
+code—think complex functions or new abstractions. Whilst it may seem like more work for you to
+adjust your pull request, the reality is your likelihood for getting review sooner shoots
+up.
+
+**Self Review Guidelines to follow**
+
+- If the PR is too large, split it if possible.
+    - Use 250 LoC (excluding test data and config changes) as a rule-of-thumb
+    - Moving and changing code should be in separate PRs or commits.
+        - Moving: Taking large code blobs and transplanting them to another file. There's generally no (or very little) actual code changed other than a cut and paste. It can even be extended to large deletions
```

Comment:
```suggestion
        - Moving: Taking large code blobs and transplanting them to another file. There's generally no (or very little) actual code changed other than a cut and paste. It can even be extended to large deletions.
```